### PR TITLE
feat(bin): improve status logs

### PR DESCRIPTION
**Pull request by shekhirin**
```console
2023-11-21T21:36:42.512280Z  INFO reth::node::events: Stage committed progress pipeline_stages=5/13 stage=Execution checkpoint=3220742 target=4739945 progress=7.97% eta=9h 50m 13s
2023-11-21T21:36:42.875589Z  INFO reth::node::events: Executing stage pipeline_stages=5/13 stage=Execution checkpoint=3220742 target=4739945 progress=7.97% eta=9h 50m 13s
2023-11-21T21:36:43.333653Z  INFO reth::cli: Status connected_peers=14 freelist=101899 stage=Execution checkpoint=3220742 target=4739945 progress=7.97% eta=9h 50m 12s
```

Two main changes:
1. Report freelist size in the "Status" log
2. Report target block number in the pipeline logs

### CODE REVIEW

File: bin/reth/src/node/events.rs [hunk 1]
```diff
@@ -26,27 +28,30 @@ const INFO_MESSAGE_INTERVAL: Duration = Duration::from_secs(25);
...
     latest_block: Option<BlockNumber>,
 }
 
+/// The stage currently being executed.
+struct CurrentStage {
+    stage_id: StageId,
+    eta: Eta,
+    checkpoint: StageCheckpoint,
+    target: Option<BlockNumber>,
+}
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:46]: move below impl

File: bin/reth/src/node/events.rs [hunk 2]
```diff
@@ -26,27 +28,30 @@ const INFO_MESSAGE_INTERVAL: Duration = Duration::from_secs(25);
 
 /// The current high-level state of the node.
 struct NodeState {
+    db: Arc<DatabaseEnv>,
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:31]: needs doc that this is used for db metrics included in logs

File: bin/reth/src/node/events.rs [hunk 3]
```diff
@@ -56,70 +61,85 @@ impl NodeState {
...
+                    current_stage.checkpoint = checkpoint;
+                    current_stage.eta.update(checkpoint);
+
+                    let target = current_stage
+                        .target
+                        .map_or("None".to_string(), |target| format!("{target}"));
+                    let progress = checkpoint
+                        .entities()
+                        .and_then(|entities| entities.fmt_percentage())
+                        .unwrap_or("None".to_string());
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:116]: nit: move `None.to_string` to helper function

File: bin/reth/src/node/events.rs [hunk 4]
```diff
@@ -26,27 +28,30 @@ const INFO_MESSAGE_INTERVAL: Duration = Duration::from_secs(25);
...
 struct NodeState {
+    db: Arc<DatabaseEnv>,
     /// Connection to the network.
     network: Option<NetworkHandle>,
     /// The stage currently being executed.
-    current_stage: Option<StageId>,
-    /// The ETA for the current stage.
-    eta: Eta,
-    /// The current checkpoint of the executing stage.
-    current_checkpoint: StageCheckpoint,
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:36]: these are now moved into a new container type CurrentStage, lgtm

File: crates/primitives/src/stage/checkpoints.rs [hunk 5]
```diff
@@ -169,9 +166,16 @@ pub struct EntitiesCheckpoint {
     pub total: u64,
 }
 
-impl Display for EntitiesCheckpoint {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{:.2}%", 100.0 * self.processed as f64 / self.total as f64)
```
Comment (mattsse) [inline crates/primitives/src/stage/checkpoints.rs:174]: removing the display is fine

**Review by mattsse — CHANGES_REQUESTED** [review]
cool!
I only have pedantic nits

File: bin/reth/src/node/events.rs (hunk 2 above)
Comment (shekhirin) [reply bin/reth/src/node/events.rs:35]: added

File: bin/reth/src/node/events.rs (hunk 1 above)
Comment (shekhirin) [reply bin/reth/src/node/events.rs]: done

**Review by mattsse — APPROVED** [review]
lgtm

File: bin/reth/src/node/events.rs [hunk 6]
```diff
@@ -189,6 +200,29 @@ impl NodeState {
     }
 }
 
+/// Helper type for formatting of optional fields:
+/// - If [Some(x)], then `x` is written
+/// - If [None], then `None` is written
+struct OptionalField<T: Display>(Option<T>);
+
+impl<T: Display> Display for OptionalField<T> {
```
Comment (gakonst) [inline bin/reth/src/node/events.rs:208]: surprised this isnt default

File: bin/reth/src/node/events.rs [hunk 7]
```diff
@@ -273,32 +308,40 @@ where
...
+                    connected_peers = this.state.num_connected_peers(),
+                    %freelist,
+                    %latest_block,
+                    "Status"
+                );
             } else {
                 info!(
                     target: "reth::cli",
                     connected_peers = this.state.num_connected_peers(),
-                    latest_block = this.state.latest_block.unwrap_or(this.state.current_checkpoint.block_number),
```
Comment (gakonst) [inline bin/reth/src/node/events.rs:301]: no need here?

**Review by gakonst — APPROVED** [review]

**Pull request merged**